envoy/server/factory_context.h

```diff
@@ -264,12 +269,44 @@ class FactoryContext : public virtual CommonFactoryContext {
   virtual ProcessContextOptRef processContext() PURE;
 };
 
+/**
+ * Context for downstream network and HTTP filters.
+ */
```

Comment:
Is this comment out of date or if it is actually the context for downstream network and http filters why does getDownstreamFactoryContext() return an OptRef? Shouldn't it always have that context?